processor/processor_test.go

```diff
@@ -3357,7 +3367,16 @@ var _ = Describe("Static Function Tests", func() {
 				},
 			}
 
-			response := getDiffMetrics("some-string-1", "some-string-2", inCountMetadataMap, inCountMap, successCountMap, failedCountMap, filteredCountMap)
+			response := getDiffMetrics(
+				"some-string-1",
+				"some-string-2",
+				inCountMetadataMap,
+				inCountMap,
+				successCountMap,
+				failedCountMap,
+				filteredCountMap,
+				stats.NOP,
+			)
```

Comment:
Should we verify the diff count as well? It should be straightforward with memstats with something like this:

```suggestion
			statsStore, err := memstats.New()
			Expect(err).To(BeNil())

			response := getDiffMetrics(
				"some-string-1",
				"some-string-2",
				inCountMetadataMap,
				inCountMap,
				successCountMap,
				failedCountMap,
				filteredCountMap,
				statsStore,
			)

			Expect(statsStore.Get("processor_diff_count", stats.Tags{
				"stage":         "some-string-2",
				"sourceId":      "some-source-id-1",
				"destinationId": "some-destination-id-1",
			}).LastValue()).To(Equal(float64(-5)))
			Expect(statsStore.Get("processor_diff_count", stats.Tags{
				"stage":         "some-string-2",
				"sourceId":      "some-source-id-2",
				"destinationId": "some-destination-id-2",
			}).LastValue()).To(Equal(float64(-7)))
```